File: hang.liu/1122.js
var relativeSortArray = function(arr1, arr2) {
  return arr1.sort((a, b) => {
    var indexA = arr2.indexOf(a);
    var indexB = arr2.indexOf(b);

    if (indexA < 0 && indexB < 0) return a - b;
    if (indexA < 0) return 1;
    if (indexB < 0) return -1;
    return indexA - indexB;
  });
};

File: xiang.li/1122.kt
class Solution {
    fun relativeSortArray(arr1: IntArray, arr2: IntArray): IntArray {
        return arr1.sortedBy {
            val index = arr2.indexOf(it)
            if (index != -1) {
                0 - arr2.size + index
            } else {
                it
            }
        }.toIntArray()
    }
}
